src/server.rs

```diff
@@ -43,16 +43,13 @@ impl Server {
         let wrq = Packet::<Wrq>::from_bytes(&buf[..nbytes]);
 
         let direction = if let Ok(rq) = rrq {
-            Some(Direction::Get(rq))
+            Direction::Get(rq)
         } else if let Ok(wq) = wrq {
-            Some(Direction::Put(wq))
+            Direction::Put(wq)
         } else {
-            None
-        };
-
-        let direction = match direction {
-            None => return Err(io::ErrorKind::InvalidInput.into()),
-            Some(d) => d,
+            let error = Packet::error(Code::IllegalOperation, "illegal operation");
```

Comment:
```suggestion
            let error = Packet::error(Code::IllegalOperation, format!("{}", Code::IllegalOperation));
```

The `Code` enum can be formatted into a String with its implementation of the `Display` trait.

Reply:
Done